applications/Fennel_Protocol.md

```diff
@@ -0,0 +1,274 @@
+| 0b. | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user can (for example) spin up one of our Substrate nodes and send test transactions, which will show how the new functionality works. |
+| 0c. | Testing Guide | Core functions will be fully covered by unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| 0d. | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 0e. | Article | We will publish an **article** that describes key generation and management with Fennel Protocol. |
+| 1. | Keypair Creation | We will integrate system-level key management libraries to provide a collection of non-blockchain key sources. We'll include AES 256 and RSA 4096 at this point. |
+| 2. | Public Key Transmission | We will develop a Substrate module that contains public key management extrinsics for broadcasting public keys' existence and locations as a signal. |
+| 3. | Public Key Retrieval | We will add an extrinsics set to the Key Management module allowing for simple retrieval of public keys given their location.<br />This will necessarily include a mechanism for verifying that the key retrieved is the key requested. |
+| 4. | Encrypted Communciation Channel Negotiation | We will create a Substrate module with extrinsics used to negotiate a three-way key handshake between two parties through cooperation of on- and off-chain compute. |
+
+### Milestone 3 — Additional features
+
+* **Estimated duration:** 1 months
+* **FTE:**  3
+* **Costs:** 20,000 USD
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| 0a. | License | Unlicense |
+| 0b. | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial** that explains how a user can (for example) spin up one of our Substrate nodes and send test transactions, which will show how the new functionality works. |
+| 0c. | Testing Guide | Core functions will be fully covered by unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| 0d. | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 0e. | Article | We will publish an **article** overviewing using Fennel Protocol in applications for secure off-chain data-centric messaging and secure on-chain signal mechanisms. |
+| 1. | Peer Retrieval | We will construct a Substrate-resident method for applications exposing the same services to identify and locate one another through on-chain signals. |
+| 2. | Encrypted Channels for IP Information | A set of off-chain workers and a parallel, scalable server will be constructed allowing for encrypted channels to be created, used, and collapsed to exchange small amounts of sensitive information such as service endpoint IP addresses. |
```

Comment:
What kind of endpoints are we talking about here? Are these channels part of the Whiteflag protocol? In the specification, it sounds more like the parachain is the communication channel.

Reply:
These are Whiteflag-parallel. The chain's communication channel allows placement of Whiteflag-specific signals for public, transparent markers, then these encrypted channels allow for more detailed communication that occurs off-chain. Consider that there's some disaster occurring in a vulnerable area - for example, a war zone, where you need to keep evacuation instructions private. You can place Whiteflag signals and use Fennel's encrypted channels and identity mechanisms to create hidden communication paths for sensitive information like that.